.github/workflows/build_linux_packages.yml

```diff
@@ -136,14 +143,13 @@ jobs:
 
       # TODO: Move to script
       - name: Upload Logs
+        if: always()
         run: |
-          aws s3 cp build/logs/ s3://therock-artifacts/${{github.run_id}}/logs/${{env.AMDGPU_FAMILIES}}/ \
-            --recursive \
-            --content-type="text/plain" \
-            --exclude "*" \
-            --include "*.log"
-          aws s3 cp build/logs/index.html s3://therock-artifacts/${{github.run_id}}/logs/${{env.AMDGPU_FAMILIES}}/
-
+          python3 build_tools/upload_logs_to_s3.py
+        env:
+          GITHUB_RUN_ID: ${{ github.run_id }}
+          AMDGPU_FAMILIES: ${{ env.AMDGPU_FAMILIES }}
+          S3_BUCKET: therock-artifacts
       - name: Add Links to Job Summary
```

Comment:
I think this step is partially needed. If you upload logs even though the build fails, you also want to add a link to the build logs (while skip adding a link to the artifacts).

Reply:
I think that should work.